pkg/xds/context/resources.go

```diff
@@ -42,6 +44,14 @@ func (r Resources) ExternalServices() *core_mesh.ExternalServiceResourceList {
 	return r.ListOrEmpty(core_mesh.ExternalServiceType).(*core_mesh.ExternalServiceResourceList)
 }
 
+func (r Resources) ExternalServicesSorted() *core_mesh.ExternalServiceResourceList {
+	ess := r.ExternalServices()
+	sort.Slice(ess.Items, func(a, b int) bool {
+		return ess.Items[a].GetMeta().GetName() < ess.Items[b].GetMeta().GetName()
+	})
+	return ess
+}
```

Comment:
I agree let's have resources in the MeshContext sorted

Reply:
Instead of `resources.Sort()` I created a new option `store.ListOrdered` and pass it every time in MeshContextBuilder. This triggers sorting in the `paginationStore`, so we don't have to code new sorting of the resources.